ddtrace/internal/sampling.py

```diff
@@ -1,7 +1,15 @@
+import json
+import os
 import re
+import sys
 from typing import Optional
 from typing import TYPE_CHECKING
 
+
+if sys.version_info >= (3, 7, 0):
+    from typing_extensions import NotRequired
+    from typing_extensions import TypedDict
```

Comment:
```suggestion
try:
   from typing import TypedDict
 except ImportError:
   from typing_extensions import TypedDict
```
TypedDict was added to [typing](https://peps.python.org/pep-0589/) in python 3.8. For `sys.version_info < (3, 8)` we should import from typing_extensions